lucene/core/src/java/org/apache/lucene/index/FieldInfo.java

```diff
@@ -130,127 +167,255 @@ public boolean checkConsistency() {
+              + io2);
     }
+  }
 
-    if (this.pointDimensionCount == 0 && dimensionCount != 0) {
-      this.pointDimensionCount = dimensionCount;
-      this.pointIndexDimensionCount = indexDimensionCount;
-      this.pointNumBytes = dimensionNumBytes;
-    } else if (dimensionCount != 0
-        && (this.pointDimensionCount != dimensionCount
-            || this.pointIndexDimensionCount != indexDimensionCount
-            || this.pointNumBytes != dimensionNumBytes)) {
+  /**
+   * Very that the provided docValues type are the same
+   *
+   * @throws IllegalArgumentException if they are not the same
+   */
+  public static void verifySameDocValuesType(
+      String fieldName, DocValuesType dv1, DocValuesType dv2) {
+    if (dv1 != dv2) {
       throw new IllegalArgumentException(
           "cannot change field \""
-              + name
+              + fieldName
+              + "\" from doc values type="
+              + dv1
+              + " to inconsistent doc values type="
+              + dv2);
+    }
+  }
+
+  /**
+   * Very that the provided doc values generations are the same
+   *
+   * @throws IllegalArgumentException if they are not the same
+   */
+  public static void verifySameDVGen(String fieldName, long dvg1, long dvg2) {
+    if (dvg1 != dvg2) {
+      throw new IllegalArgumentException(
+          "cannot change field \""
+              + fieldName
+              + "\" from doc values generation="
+              + dvg1
+              + " to inconsistent doc values generation="
+              + dvg2);
+    }
+  }
+
+  /**
+   * Very that the provided posting options are the same
+   *
+   * @throws IllegalArgumentException if they are not the same
+   */
+  public static void verifySamePostingsOptions(
+      String fieldName,
+      boolean stv1,
+      boolean on1,
+      boolean sp1,
+      boolean stv2,
+      boolean on2,
+      boolean sp2) {
+    if (stv1 != stv2 || on1 != on2 || sp1 != sp2) {
+      throw new IllegalArgumentException(
+          "cannot change field \""
+              + fieldName
+              + "\" from storeTermVector="
+              + stv1
+              + ", omitNorms="
+              + on1
+              + ", storePayloads="
+              + sp1
+              + " to inconsistent storeTermVector="
+              + stv2
+              + ", omitNorms="
+              + on2
+              + ", storePayloads="
+              + sp2);
+    }
+  }
+
+  /**
+   * Very that the provided points indexing options are the same
+   *
+   * @throws IllegalArgumentException if they are not the same
+   */
+  public static void verifySamePointsOptions(
+      String fieldName, int dc1, int idc1, int nb1, int dc2, int idc2, int nb2) {
+    if (dc1 != dc2 || idc1 != idc2 || nb1 != nb2) {
+      throw new IllegalArgumentException(
+          "cannot change field \""
+              + fieldName
               + "\" from points dimensionCount="
-              + this.pointDimensionCount
+              + dc1
```

Comment:
I liked the descriptive variable names better.

Reply:
Addressed in 6e7540ebd0ef79536cffabcf0ddc7a592b792252